test/integration/api/api.test.js

```diff
@@ -0,0 +1,467 @@
+				});	
+			},
+			function getDummySchedule(callback){
+				var getSchedulesOptions={
+						url:schedulerURI+'/v2/schedules/dummy',
+						method:'GET',
+						json:true
+				};
+				request(getSchedulesOptions,function(error,response,body){
+					expect(error).to.be.null;
+					expect(response.statusCode).to.equal(200);
+					expect(response.body.schedules.specific_date).to.not.be.empty;
+					expect(response.body.schedules.recurring_schedule).to.not.be.empty;
+					expect(response.body.schedules.recurring_schedule).to.have.lengthOf(4);
+					expect(response.body.schedules.specific_date).to.have.lengthOf(2);
+					callback(error,response);
+				});
+			}
+			],done);
+		});
+
+	});
+	
+	context('Failure scenarios',function(){
+		// Cleanup the DB before running every context
+		before(function(done){
+			var deletePolicyOptions={
+					url:apiServerURI+'/v1/policies/dummy',
+					method:'DELETE',
+					json:true
+			};
+			request(deletePolicyOptions,function(error,response,body){
+				done();
+			});		
+		});	
+
+		it('Should fail to create policy and associated schedules due to some validation error with the policy',function(done){
+			var newFakePolicy=JSON.parse(JSON.stringify(fakePolicy));
+			newFakePolicy.instance_min_count=5;
+			async.series([	
+			function createDummyPolicy(callback){
+				var createOptions={
+						url:apiServerURI+'/v1/policies/dummy',
+						method:'PUT',
+						body:newFakePolicy,
+						json:true
+				};
+				request(createOptions,function(error,response,body){
+					expect(error).to.be.null;
+					expect(response.statusCode).to.equal(400);
+					callback(error,response);
+				});
+			},
+			function getDummyPolicy(callback){
+				var policyOptions={
+						url:apiServerURI+'/v1/policies/dummy',
+						method:'GET',
+						json:true
+				};
+				request(policyOptions,function(error,response,body){
+					expect(error).to.be.null;
+					expect(response.statusCode).to.equal(404);
+					callback(error,response);
+				});	
+			},
+			function getDummySchedule(callback){
+				var getSchedulesOptions={
+						url:schedulerURI+'/v2/schedules/dummy',
+						method:'GET',
+						json:true
+				};
+				request(getSchedulesOptions,function(error,response,body){
+					expect(error).to.be.null;
+					expect(response.statusCode).to.equal(404);
+					callback(error,response);
+				});
+			}
+			],done);
+		});	
+		
+		it('Should return a 404 status code while deleting a non-existing policy and associated schedules',function(done){
+				var deletePolicyOptions={
+						url:apiServerURI+'/v1/policies/dummy',
+						method:'DELETE',
+						json:true
+				};
+				request(deletePolicyOptions,function(error,response,body){
+					expect(response.statusCode).to.equal(404);
+					expect(response.body.success).to.equal(false);
+					expect(response.body.error.statusCode).to.equal(404);
+					done();
+				});
+		});	
+		
+		it('Should return a 404 status code while retrieving a non-existing policy',function(done){
```

Comment:
Didn't add explicit tests for this. Already tested as part of the existing tests.